pycaret/internal/pycaret_experiment/pycaret_experiment.py

```diff
@@ -447,287 +530,46 @@ def pull(self, pop=False) -> pd.DataFrame:  # added in pycaret==2.2.0
-                return self.train_transformed
-        else:
-            # Use fully trained pipeline to get the requested data
-            return pd.concat(
-                [
-                    *_pipeline_transform(
-                        pipeline=self.pipeline_fully_trained, y=self.y, X=self.X
-                    )
-                ],
-                axis=1,
-            )
+    def X_train_transformed(self):
+        """Transformed feature set of the training set."""
+        return self.pipeline.transform(self.X_train, filter_train_only=False)
 
     @property
     def train_transformed(self):
         """Transformed training set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            if self.target_param:
-                return pd.concat(
-                    [self.X_train_transformed, self.y_train_transformed],
-                    axis=1,
-                )
-            else:
-                return self.X_train_transformed
-        else:
-            # Use pipeline trained on training data only to get the requested data
-            # In time series, the order of arguments and returns may be reversed.
-            return pd.concat(
-                [
-                    *_pipeline_transform(
-                        pipeline=self.pipeline, y=self.y_train, X=self.X_train
-                    )
-                ],
-                axis=1,
-            )
-
-    @property
-    def test_transformed(self):
-        """Transformed test set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            return pd.concat(
-                [self.X_test_transformed, self.y_test_transformed],
-                axis=1,
-            )
-        else:
-            # When transforming the test set, we can and should use all data before that
-            # In time series, the order of arguments and returns may be reversed.
-            all_data = pd.concat(
-                [
-                    *_pipeline_transform(
-                        pipeline=self.pipeline_fully_trained,
-                        y=self.y,
-                        X=self.X,
-                    )
-                ],
-                axis=1,
-            )
-            # Return the y_test indices not X_test indices.
-            # X_test indices are expanded indices for handling FH with gaps.
-            # But if we return X_test indices, then we will get expanded test
-            # indices even for univariate time series without exogenous variables
-            # which would be confusing. Hence, we return y_test indices here and if
-            # we want to get X_test indices, then we use self.X_test directly.
-            # Refer:
-            # https://github.com/sktime/sktime/issues/2598#issuecomment-1203308542
-            # https://github.com/sktime/sktime/blob/4164639e1c521b112711c045d0f7e63013c1e4eb/sktime/forecasting/model_evaluation/_functions.py#L196
-            return all_data.loc[self.idx[1]]
+        return self.X_train_transformed
```

Comment:
What about y values here?

Reply:
this is overriden in `SupervisedExperiment` to return only X values